src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java

```diff
@@ -83,39 +85,58 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDeta
 
     }
 
-    // Helper method to create term queries with variations for spaces
-    private BoolQueryBuilder createNameMatchQuery(String field, String name) {
-        return QueryBuilders.boolQuery()
-            .should(QueryBuilders.termQuery(field, name)) // Exact match
-            .should(QueryBuilders.termQuery(field, " " + name)) // Prepend space
-            .should(QueryBuilders.termQuery(field, name + " ")) // Append space
-            .should(QueryBuilders.termQuery(field, " " + name + " ")); // Prepend and append space
+    public String[] normalizeAndSplit(String name) {
+        // remove brackets and anything inside them
+        String nameWithoutStuffInBrackets = name.replaceAll("\\([^)]*\\)", "");
+
+        // replace multiple consecutive "/" with a single "/" because we might have multiple names due to translations
+        String nameWithoutExtraSlashes = nameWithoutStuffInBrackets.replaceAll("/+", "/");
+
+        // remove illegal characters that we've spotted in prod data
+        String illegalCharacters = ".=!*_";
```

Comment:
There were some other punctuation characters in the DB - did you find it's safe to leave them in?
```
"-"
"'"
""""
"/"
"`"
"("
":"
"!"
","
"|"
" ​"
"_"
"]"
"="
")"
"."
"*"
```

Reply:
If they've put something in brackets, I'm assuming it isn't the name and the brackets and what's in the brackets gets removed.
The / seems to be to split alternative spellings - this is a bit of a weird one because they are meant to have exactly what is on the certificate which obvs isn't going to be 2 names, but I've gone with splitting on / and then doing extra searches on the array of names. Just based on the example you gave.
I thought I'd covered what you gave me but must not have if '`-_| are also in there. I did get my prod access renewed now tho. Maybe I need to do some more searching?
Trying to think if folk can have `' in name, I think they can have - in their name.

The alternative is to do fuzzy search but that would give more false positives I think.